python/sglang/backend/runtime_endpoint.py

```diff
@@ -73,9 +73,10 @@ def cache_prefix(self, prefix_str: str):
         assert res.status_code == 200
 
     def commit_lazy_operations(self, s: StreamExecutor):
+        lazy_text = s.text_.replace("<image>", "") if len(s.images_)>0 else s.text_
         res = http_request(
             self.base_url + "/generate",
-            json={"text": s.text_, "sampling_params": {"max_new_tokens": 0}},
+            json={"text": lazy_text, "sampling_params": {"max_new_tokens": 0}},
```

Comment:
This is not 100% correct. Could you try this?

```
        data = {"text": s.text_, "sampling_params": {"max_new_tokens": 0}}
        self._add_images(s, data)
        res = http_request(
            self.base_url + "/generate",
            json=data,
            auth_token=self.auth_token,
            api_key=self.api_key,
            verify=self.verify,
        )
        assert res.status_code == 200
```

Reply:
Thanks! It works for me.